crates/sui-indexer/src/handlers/event_handler.rs

```diff
@@ -44,22 +43,26 @@ impl EventHandler {
         let mut pg_pool_conn = get_pg_pool_connection(self.pg_connection_pool.clone())?;
 
         let mut next_cursor = None;
-        let event_log = read_event_log(&mut pg_pool_conn)?;
-        let (tx_dig_opt, event_seq_opt) = (
-            event_log.next_cursor_tx_dig,
-            event_log.next_cursor_event_seq,
-        );
-        if let (Some(tx_dig), Some(event_seq)) = (tx_dig_opt, event_seq_opt) {
-            let tx_digest = tx_dig.parse().map_err(|e| {
-                IndexerError::TransactionDigestParsingError(format!(
-                    "Failed parsing transaction digest {:?} with error: {:?}",
-                    tx_dig, e
-                ))
-            })?;
-            next_cursor = Some(EventID {
-                tx_digest,
-                event_seq,
-            });
+        let last_event_opt = read_last_event(&mut pg_pool_conn)?;
+        if let Some(last_event) = last_event_opt {
```

Comment:
use match using tuple `(last_event_opt, last_event.next_cursor_transaction_digest, last_event.next_cursor_event_sequence)` might be cleaner then nested ifs

Reply:
next_cursor_transaction_digest and next_cursor_event_sequence are parallel indeed, changed that to match.
I saw `if let(...) && let(...)` are available but not stable, would be handy when stable.